rollout/sync.go

```diff
@@ -617,6 +617,11 @@ func (c *rolloutContext) calculateRolloutConditions(newStatus v1alpha1.RolloutSt
 	// Check for progress. Only do this if the latest rollout hasn't completed yet and it is not aborted
 	if !isCompleteRollout && !isAborted {
 		switch {
+		case conditions.RolloutTimedOut(c.rollout, &newStatus) && defaults.GetAbortExceedProgressDeadlineOrDefault(c.rollout):
+			msg := fmt.Sprintf(conditions.RolloutTimeOutMessage, c.rollout.Name)
+			c.pauseContext.AddAbort(msg)
+			abortCondition := conditions.NewRolloutCondition(v1alpha1.RolloutPaused, corev1.ConditionTrue, conditions.TimedOutReason, msg)
+			conditions.SetRolloutCondition(&newStatus, *abortCondition)
```

Comment:
Shouldn't this logic go into the latter case statement:

```golang
case !isIndefiniteStep(c.rollout) && conditions.RolloutTimedOut(c.rollout, &newStatus):
```

We do not want to abort if we are at an indefinite pause step.

Reply:
The logic is moved to the suggested case. Please take another look. Thanks.